openedx/core/djangoapps/programs/utils.py

```diff
@@ -82,6 +100,11 @@ def get_program_urls(program_data):
         'commerce_api_url': reverse('commerce_api:v0:baskets:create'),
         'buy_button_url': ecommerce_service.get_checkout_page_url(*skus),
         'program_record_url': program_record_url,
+        'buy_subscription_url': get_buy_subscription_url(
+            *skus, program_uuid=program_uuid) if subscription_eligible else None,
+        'manage_subscription_url': settings.ORDER_HISTORY_MICROFRONTEND_URL if subscription_eligible else None,
+        'subscriptions_learner_help_center_url': (settings.SUBSCRIPTIONS_LEARNER_HELP_CENTER_URL
+                                                  if subscription_eligible else None,)
```

Comment:
This feels like we should break this into its own chunk of code. We can _update_ the `urls`. 

We can do something like...
```
urls = {
    <blah>
}

if subscription_eligible:
   urls['buy_subscription_url'] = get_buy_subscription_url(*skus, program_uuid=program_uuid)
   urls['manage_subscription_url'] = settings.ORDER_HISTORY_MICROFRONTEND_URL
   urls['subscriptions_learner_help_center_url'] = settings.SUBSCRIPTIONS_LEARNER_HELP_CENTER_URL
```
This seems to be cleaner than repeating the same ternary expression based on the value of `subscription_eligible`.

Reply:
updating the urls based on the flag would have made the context keys inconsistent and frontend would have to take extra care to check for the keys in that case. That was the reason I went with the above approach at first. But I have removed the conditionals all together now and we will send the urls in any case. We will leave it up to frontend now to render and use the URLS based on the whether it is subscriptions UI or not.